x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/jwt/JwtRealmSettings.java

```diff
@@ -0,0 +1,310 @@
+        set.addAll(RealmSettings.getStandardSettings(TYPE));
+        // JWT Issuer settings
+        set.addAll(List.of(ALLOWED_ISSUER, ALLOWED_SIGNATURE_ALGORITHMS, ALLOWED_CLOCK_SKEW, JWKSET_PATH));
+        // JWT Audience settings
+        set.addAll(List.of(ALLOWED_AUDIENCES));
+        // JWT End-user settings
+        set.addAll(
+            List.of(
+                PRINCIPAL_CLAIM.getClaim(),
+                PRINCIPAL_CLAIM.getPattern(),
+                GROUPS_CLAIM.getClaim(),
+                GROUPS_CLAIM.getPattern(),
+                POPULATE_USER_METADATA
+            )
+        );
+        // Client TLS settings for incoming connections (subset of SSLConfigurationSettings)
+        set.addAll(
+            List.of(
+                CLIENT_AUTHENTICATION_TYPE,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PATH,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_LEGACY_TRUSTSTORE_PASSWORD,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_ALGORITHM,
+                CLIENT_AUTHENTICATION_TRUSTSTORE_TYPE,
+                CLIENT_AUTHENTICATION_CERT_AUTH_PATH
+            )
+        );
+        // Delegated authorization settings: authorization_realms
+        set.addAll(DelegatedAuthorizationSettings.getSettings(TYPE));
+        // JWT Cache settings
+        set.addAll(List.of(CACHE_TTL, CACHE_MAX_USERS));
+        // Standard HTTP settings for outgoing connections to get JWT issuer jwkset_path
+        set.addAll(
+            List.of(
+                HTTP_CONNECT_TIMEOUT,
+                HTTP_CONNECTION_READ_TIMEOUT,
+                HTTP_SOCKET_TIMEOUT,
+                HTTP_MAX_CONNECTIONS,
+                HTTP_MAX_ENDPOINT_CONNECTIONS,
+                HTTP_PROXY_HOST,
+                HTTP_PROXY_PORT,
+                HTTP_PROXY_SCHEME
+            )
+        );
+        // Standard TLS connection settings for outgoing connections to get JWT issuer jwkset_path
+        set.addAll(SSLConfigurationSettings.getRealmSettings(TYPE));
+        return set;
+    }
+
+    // JWT issuer settings
+
+    public static final Setting.AffixSetting<String> ALLOWED_ISSUER = RealmSettings.simpleString(
+        TYPE,
+        "allowed_issuer",
+        Setting.Property.NodeScope
+    );
+    public static final Setting.AffixSetting<TimeValue> ALLOWED_CLOCK_SKEW = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "allowed_clock_skew",
+        key -> Setting.timeSetting(key, DEFAULT_ALLOWED_CLOCK_SKEW, Setting.Property.NodeScope)
+    );
+    public static final Setting.AffixSetting<List<String>> ALLOWED_SIGNATURE_ALGORITHMS = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        "allowed_signature_algorithms",
+        key -> Setting.listSetting(key, DEFAULT_ALLOWED_SIGNATURE_ALGORITHMS, Function.identity(), v -> {
+            if (SUPPORTED_SIGNATURE_ALGORITHMS.contains(v) == false) {
```

Comment:
This setting was copied from OIDC. IntelliJ warns there might be a problem with the copied code.
- Warning:(143, 57) 'List<String>' may not contain objects of type 'List<String>'

Reply:
> This setting was copied from OIDC.

The setting in OIDC is a `Setting.AffixSetting<String>` but you changed it to `Setting.AffixSetting<List<String>>` and this is what Intellij warns you about. 

Do we really need to support multiple signing algorithms per realm ? I presume that since we support a single Issuer we should be fine with supporting a single algorithm ( this is the case in OIDC too )
If we want to support multiple algos ( I guess there might be a case where different RPs are configured with different algorithms and we need to validate JWTs for both  ) then we need to change this implementation: 

`v` is a list and you need to check that the contents of `v` are a subset of `SUPPORTED_SIGNATURE_ALGORITHMS`